src/main/java/de/neemann/digital/core/io/PortSocket.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2019 Helmut Neemann & Mats Engstrom.
+ * Use of this source code is governed by the GPL v3 license
+ * that can be found in the LICENSE file.
+ */
+package de.neemann.digital.core.io;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+//import java.net.SocketTimeoutException;
+//import java.util.concurrent.TimeoutException;
+
+/**
+ *
+ */
+public class PortSocket extends Thread {
+
+
+
+    private Port port;
+    private boolean telnetMode;
+    private ServerSocket serverSocket;
+    private DataOutputStream outStream;
+
+    /**
+     * @param port instance of creator
+     * @param listenPort - TCP port to liste to
+     * @param telnetMode - True if a telnet terminal will connect
+     */
+    public PortSocket(Port port, int listenPort, boolean telnetMode) {
+        this.port = port;
+        this.telnetMode = telnetMode;
+        try {
```

Comment:
Add the `setDaemon(true);` call.

Reply:
Ok, added. And after reading the dox it seems like threads started by a daemon thread inherits the status so I don't need to daemonize the actual comms thread after that.